File: go/src/product.go
package shoppingcart

import "errors"

type Product struct {
	code string
}

func (p *Product) Name() (string, error) {
	switch p.code {
	case "apple":
		return "りんご", nil
	case "grape":
		return "ぶどう", nil
	case "orange":
		return "みかん", nil
	}
	return "", errors.New("code値が不正です")
}

func (p *Product) Price() (int, error) {
	switch p.code {
	case "apple":
		return 50, nil
	case "grape":
		return 100, nil
	case "orange":
		return 70, nil
	}
	return 0, errors.New("code値が不正です")
}

File: typescript/src/product.ts
export class Product {
  /**
   * 商品名
   * @var {string}
   */
  public name: string;

  /**
   * 単価
   * @var {number}
   */
  public price: number;

  /**
   * @param {string} code
   */
  constructor(public code: string) {
    // コード値に応じた商品情報を格納する
    switch (code) {
      case "grape":
        this.name = "ぶどう";
        this.price = 100;
        break;
      case "apple":
        this.name = "りんご";
        this.price = 50;
        break;
      case "orange":
        this.name = "みかん";
        this.price = 70;
        break;
      default:
        throw new Error("codeが不正な値です");
    }
  }
}
